cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/OAuthHandler.java

```diff
@@ -265,24 +271,18 @@ private boolean checkCredIsValid(HttpResponse response) throws OAuthServiceExcep
       throw new OAuthServiceException(HttpURLConnection.HTTP_INTERNAL_ERROR, "Failed to parse JSON", e);
     }
 
-    if (refreshTokenResponse.getAccessToken() == null || refreshTokenResponse.getAccessToken().isEmpty()) {
-      return false;
-    }
-    return true;
+    return refreshTokenResponse.getAccessToken() != null &&
+        !refreshTokenResponse.getAccessToken().isEmpty();
   }
 
   private HttpRequest createGetRefreshTokenRequest(OAuthProvider provider, String code, String redirectURI)
       throws OAuthServiceException {
     OAuthClientCredentials clientCreds = provider.getClientCredentials();
     try {
-      return HttpRequest.post(new URL(
-          String.format("%s?code=%s&grant_type=authorization_code&redirect_uri=%s",
-              provider.getTokenRefreshURL(),
-              code,
-              redirectURI)))
-          .addHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED)
```

Comment:
Generally for any body in HTTP you should specify content-type. Could you clarify why are you removing it?

Reply:
It was an unexpected way of using that header in [Salesforce Request an Access Token](https://help.salesforce.com/s/articleView?id=sf.remoteaccess_oauth_web_server_flow.htm&type=5) that got me confused. I've put it back and everything seems to be working fine.